### PR TITLE
Send new questions to slack channel

### PR DESCRIPTION
**Pull request by Errorname**
Any new questions will be posted in a slack channel ([#faq](https://zenika.slack.com/messages/CA5MDJHGE))

Before merging:
* [x] Write documentation [slack_hook env + faq_url env] (wait the new doc #44)

### CODE REVIEW

File: server/src/slack/newQuestion.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  // Ascii emoticons to :emoji:
+  return Object.keys(emoticons).reduce((text, emoticon) => {
+    return text.split(emoticon + ' ').join(emoticons[emoticon] + ' ')
+  }, text)
+}
+
+export default async event => {
+  const channel_hook = process.env['SLACK_CHANNEL_HOOK']
```
Comment (hgwood) [inline server/src/slack/newQuestion.js]: Should be `channelHook`, no?

File: server/src/slack/newQuestion.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  const response = await fetch(channel_hook, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(message)
+  })
+
+  return { data: { response: response.text() } }
```
Comment (hgwood) [inline server/src/slack/newQuestion.js]: `response.text()` returns a promise, doesn't it? Is this what you expect?

Comment (Errorname) [reply server/src/slack/newQuestion.js]: No, it's not what I wanted, thanks for noticing

File: server/src/slack/newQuestion.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  const channelHook = process.env['SLACK_CHANNEL_HOOK']
+
+  if (!channelHook) {
+    console.log('Please provide a valid slack channel hook!')
+    return { error: 'Module not configured correctly.' }
+  }
+
+  const question = event.data.Question.node
+
+  const url = `https://faq.zenika.com/q/${question.slug}-${question.id}`
```
Comment (bpetetot) [inline server/src/slack/newQuestion.js]: May the server URL should be also managed in env file ?

Comment (Errorname) [reply server/src/slack/newQuestion.js]: Indeed it should

File: docs/integrations.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+Slack is a collaboration chat tool often used in and out of organizations to help team communicate and coordinate in a more effective manner.
+
+FAQ has the following integration with slack:
+
+* Sends new questions into a dedicated slack channel
+
+### Configuration
+
+In order to integrate with Slack, you will need to create an application. It can easily be done using the [Slack "Build" dashboard](https://api.slack.com/). Follow the instructions from Slack to create your app. See below for configuration for your environment variables.
```
Comment (hgwood) [inline docs/integrations.md]: You can drop "easily", it's relative. Easy for someone might be easy for others.

File: docs/integrations.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+### Environment variables
+
+#### `SLACK_CHANNEL_HOOK`
+
+_Used by: Back-end_
+
+This variable is a hook which is called by the backend in order to send the message to the channel. You can configure a hook on the [Slack "Build" dashboard](https://api.slack.com/). Choose your app, then go to "Features > Incoming Webhooks". Active the feature and create a hook for the dedicated channel you wish to send questions to. The generated url is what you need for this variable.
```
Comment (hgwood) [inline docs/integrations.md]: I suggest "The variable is a hook URI..." so it is more obvious what the format is.

File: docs/integrations.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+Workplace is a collaborative platform run by Facebook and can be labeled as "Facebook for companies".
+
+FAQ has the following integration with Workplace:
+
+* Button to share a question on Workplace
+
+### Configuration
+
+There is no configuration needed for workplace to work.
```
Comment (bpetetot) [inline docs/integrations.md]: I'm confused, I don't understand how FAQ can be integrated with Workplace without any configuration in Workplace or in FAQ app. 😕

Comment (Errorname) [reply docs/integrations.md]: To share with Worplace, we only need to open a popup with the following URL: `https://work.facebook.com/sharer.php?display=popup&u=[the_url_to_share]`. There is no key needed, nor configuration beforehand to make this work.

Should I had a sentence in the documentation to clear this confusion ?

**Pull request merged**